edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -0,0 +1,937 @@
+            if (string.Equals(customObject.Metadata.Name, this.resourceName, StringComparison.OrdinalIgnoreCase))
+            {
+                using (await this.watchLock.LockAsync())
+                {
+                    V1ServiceList currentServices = await this.client.ListNamespacedServiceAsync(this.k8sNamespace, labelSelector: this.deploymentSelector);
+                    V1DeploymentList currentDeployments = await this.client.ListNamespacedDeploymentAsync(this.k8sNamespace, labelSelector: this.deploymentSelector);
+                    Events.DeploymentStatus(type, this.resourceName);
+                    switch (type)
+                    {
+                        case WatchEventType.Added:
+                        case WatchEventType.Modified:
+                            this.ManageDeployments(currentServices, currentDeployments, customObject);
+                            break;
+
+                        case WatchEventType.Deleted:
+                            {
+                                // Delete the deployment.
+                                // Delete any services.
+                                IEnumerable<Task<V1Status>> removeServiceTasks = currentServices.Items.Select(i => this.client.DeleteNamespacedServiceAsync(i.Metadata.Name, this.k8sNamespace, new V1DeleteOptions()));
+                                await Task.WhenAll(removeServiceTasks);
+                                IEnumerable<Task<V1Status>> removeDeploymentTasks = currentDeployments.Items.Select(
+                                    d => this.client.DeleteNamespacedDeployment1Async(
+                                        d.Metadata.Name,
+                                        this.k8sNamespace,
+                                        new V1DeleteOptions(propagationPolicy: "Foreground"),
+                                        propagationPolicy: "Foreground"));
+                                await Task.WhenAll(removeDeploymentTasks);
+                                this.currentModules = ModuleSet.Empty;
+                            }
+
+                            break;
+                        case WatchEventType.Error:
+                            Events.DeploymentError();
+                            break;
+                    }
+                }
+            }
+            else
+            {
+                Events.DeploymentNameMismatch(customObject.Metadata.Name, this.resourceName);
+            }
+        }
+
+        private string DeploymentName(string moduleId) => KubeUtils.SanitizeK8sValue(moduleId);
+
+        private async void ManageDeployments(V1ServiceList currentServices, V1DeploymentList currentDeployments, EdgeDeploymentDefinition<TConfig> customObject)
+        {
+            var desiredModules = ModuleSet.Create(customObject.Spec.ToArray());
+            var moduleIdentities = await this.moduleIdentityLifecycleManager.GetModuleIdentitiesAsync(desiredModules, this.currentModules);
```

Comment:
Use the new bool parameter here.

Reply:
@darobs  Doh!! I had that in an earlier revision? Added back.